app/javascript/gobierto_visualizations/webapp/containers/contracts/EntityTreeMapNested.vue

```diff
@@ -183,20 +183,11 @@ export default {
 
         } else if (depth === 2) {
           let totalContracts = d.children === undefined ? '' : d.children
-          let contractsString = ''
-          if (totalContracts) {
-            totalContracts = totalContracts.filter(contract => typeof contract.data !== "function")
-            while (i < totalContracts.length) {
-              let contractAmount = selected_size === 'final_amount_no_taxes' ? `${totalContracts[i].data.children[0].value}` : `${totalContracts[i].data.children[0].final_amount_no_taxes}`
-              contractsString = `${contractsString}
-              <div class="depth-second-container">
-                <p class="depth-second-title">${totalContracts[i].data.children[0].title}</p>
-                <p class="text-depth-third">${I18n.t('gobierto_visualizations.visualizations.contracts.contract_amount')}: <b>${money(contractAmount)}</b></p>
-                <p class="text-depth-third">${I18n.t('gobierto_visualizations.visualizations.contracts.tender_amount')}: <b>${money(totalContracts[i].data.children[0].initial_amount_no_taxes)}</b></p>
-              </div>`
-              i++
-            }
-          }
+          let contractsString = totalContracts
```

Comment:
Are so many the differences between `EntityTreeMapNested.vue` and `CategoriesTreeMapNested.vue` to keep two different files with some much similar code?

Reply:
At least reuse some common functions in a vue-mixin....